src/server/common.h

```diff
@@ -371,9 +371,11 @@ class ThreadLocalMutex {
   void unlock();
 
  private:
+  const bool big_value_enabled_;
```

Comment:
it will also remove the need for having a class level variable. just use a global variable inside lock and unlock.